marker/builders/structure.py

```diff
@@ -45,27 +45,21 @@ def group_caption_blocks(self, page: PageGroup):
 
             block_structure = [block_id]
             selected_polygons = [block.polygon]
-            for j, prev_block_id in enumerate(page.structure[:i][::-1]):
```

Comment:
This changes the logic - the idea was to be able to merge multiple captions on each side